### DISCUSSION TITLE
How to integrate the admin menu in a custom module and make it available for a logged in user only

### DISCUSSION THREAD

**Discussion by MarcBruins** [OrchardCMS/OrchardCore#6901] [Q&A] [answered]
So I've followed the [Add an admin menu](https://docs.orchardcore.net/en/dev/docs/guides/add-admin-menu/) steps. Thanks to that and the input in another discussion I successfully created my custom module with it's own logic.

When i start the CMS and navigate to my custom module through the menu the menu disappears. My first question is, how do i include the admin menu, so that i can navigate easily from and to my custom module?

The second question is, how can i make sure that only a admin or editor can see the module? Right now a user that is not logged in can directly navigate to the page and use the custom module.

**Comment by Piedone** [comment]
> When i start the CMS and navigate to my custom module through the menu the menu disappears. My first question is, how do i include the admin menu, so that i can navigate easily from and to my custom module?

I don't really understand, can you elaborate? When does your menu disappear exactly? Does that mean that you see it once though?

You can tie menu items to permissions [like this](https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/dev/Navigation/PersonsAdminMenu.cs#L67), and then you can grant only the Editor role that permission. For this, you can use one of the built-in permissions too but probably a better idea is to add your own into your module [like this](https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/dev/Permissions/PersonPermissions.cs).

Do note that authorizing the menu item is just that: It'll hide the menu for unauthorized users but if they know the URL of whatever the menu points to, they'll still be able to visit it. If you want authorization in your own controller, for example, do it [like this](https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/421bf42236f0ec5766be3caedfadd6ce169e769e/Controllers/AuthorizationController.cs#L79).

**Reply by MarcBruins** [reply]
What i mean is that as soon as i login, i see the admin menu. Everything is there and works perfectly, now as soon as i navigate to my own module, by clicking on it in the admin menu. The menu is gone and it only renders the html/css in my module. The desired state is that it’s my own html/css/logic and the css and the admin menu from Orchard.

Thanks for showing how to deal with the permissions, this answers that question 👍 I’ll have a go at it next Monday

**Reply by deanmarcussen** [reply]
You need to either call your controller `AdminController` or apply the `AdminAttribute` to the controller, so `[Admin]`

**Comment by MarcBruins** [comment, marked as answer]
Thanks guys, once again you are such great help. I’ve created the custom permissions and added the `admin` attribute to the controller and that worked perfectly.
